src/main/scala/org/apache/spark/sql/delta/commands/MergeIntoCommand.scala

```diff
@@ -562,11 +573,12 @@ case class MergeIntoCommand(
     // create an alias
     val aliases = plan.output.map {
       case newAttrib: AttributeReference =>
-        val existingTargetAttrib = getTargetOutputCols(deltaTxn).find(_.name == newAttrib.name)
+        val targetOutputColsMap = getTargetOutputColsMap(deltaTxn)
```

Comment:
arent you still calling this function to generate a map for each every column in the plan.output? so isnt this still quadratic??